pkg/meta/tkv.go

```diff
@@ -3756,3 +3758,146 @@ func (m *kvMeta) loadDumpedACLs(ctx Context) error {
 		return nil
 	})
 }
+
+type kvEntryScanner struct {
+	cache       []*Entry
+	cnt         int
+	m           *kvMeta
+	defaultCap  int
+	parentInode Ino
+	plus        bool
+	bgChan      chan []*Entry
+}
+
+func (m *kvMeta) NewMetaEntryScanner(inode Ino, plus bool) EntryScanner {
+	c := &kvEntryScanner{
+		cache:       make([]*Entry, 0),
+		cnt:         0,
+		m:           m,
+		parentInode: inode,
+		defaultCap:  DefaultCap,
+		plus:        plus,
+		bgChan:      nil,
+	}
+	return c
+}
+
+func (sc *kvEntryScanner) Valid() bool {
+	return true
+}
+
+func (sc *kvEntryScanner) Close() {
+	if sc.bgChan != nil {
+		_ = <-sc.bgChan
+		sc.cache = make([]*Entry, 0)
+		sc.cnt = 0
+		sc.bgChan = nil
+	}
+}
+
+func (sc *kvEntryScanner) scanEntries(startKey, endKey []byte, limit int) ([]*Entry, error) {
+	if limit <= 0 {
+		return nil, nil
+	}
+	entries := make([]*Entry, 0)
+	prefix := len(sc.m.entryKey(sc.parentInode, ""))
+	err := sc.m.client.txn(func(tx *kvTxn) error {
+		var cnt int = 0
+		tx.scan(startKey, endKey, false, func(name, buf []byte) bool {
+
+			typ, ino := sc.m.parseEntry(buf)
+			if len(name) == prefix {
+				logger.Errorf("Corrupt entry with empty name: inode %d parent %d", ino, sc.parentInode)
+				return false
+			}
+			entries = append(entries, &Entry{
+				Inode: ino,
+				Name:  []byte(name)[prefix:],
+				Attr:  &Attr{Typ: typ},
+			})
+			cnt++
+			return cnt < limit
+		})
+		return nil
+	}, 0)
+	return entries, err
+}
+
+func (sc *kvEntryScanner) fillAttr(entries []*Entry) (err error) {
+	if len(entries) == 0 {
+		return nil
+	}
+	var keys = make([][]byte, len(entries))
+	for i, e := range entries {
+		keys[i] = sc.m.inodeKey(e.Inode)
+	}
+	var rs [][]byte
+	err = sc.m.client.txn(func(tx *kvTxn) error {
+		rs = tx.gets(keys...)
+		return nil
+	}, 0)
+	if err != nil {
+		return err
+	}
+	for j, re := range rs {
+		if re != nil {
+			sc.m.parseAttr(re, entries[j].Attr)
+		}
+	}
+	return err
+}
+
+func (sc *kvEntryScanner) GetData(start int) ([]*Entry, syscall.Errno) {
+	if sc.bgChan == nil {
+		sc.bgChan = make(chan []*Entry, 1)
+		go sc.fetchEntries()
```

Comment:
Maybe we should conduct a performance benchmark test to confirm if there is a need to control the number of goroutines. 
In user scenarios, the creation of an excessive number of goroutines in this section could potentially cause a decline in performance, which may also affect other APIs within the system.

Reply:
Sorry for late reply. I have already rewritten this piece of code using the waitgroup to synchronize. Now, the background goroutines will not remain alive once they have completed their data-fetching tasks. Could you please confirm if this modification is acceptable?